jetbot/bot.py

```diff
@@ -513,27 +515,7 @@ async def channels(ctx: commands.Context):
                                 await channel.set_permissions(overwrite, overwrite=None)
                             await channel.edit(slowmode_delay=5, sync_permissions=True)
 
-                    # Handle legacy single-channel configuration
-                    elif item.get("channel_name"):
-                        channel = await self.product_channel(item["channel_name"], item["category_name"])
-                        if not channel:
-                            print("Creating channel... " + item["channel_name"])
-                            channel = await guild.create_text_channel(item["channel_name"], category=category)
-                            new.append(channel)
-
-                        # Set permissions
-                        print("Updating channel... " + item["channel_name"] + " in " + item["category_name"])
-                        for overwrite in channel.overwrites:
-                            await channel.set_permissions(overwrite, overwrite=None)
-                        await channel.edit(slowmode_delay=5, sync_permissions=True)
-
-                    channels_to_update = []
-                    if "channels" in item:
-                        channels_to_update.extend(await self.product_channels(item))
-                    elif item.get("channel_name"):
-                        channel = await self.product_channel(item["channel_name"], item["category_name"])
-                        if channel:
-                            channels_to_update.append(channel)
+                    channels_to_update = await self.product_channels(item)
 
                     for channel in channels_to_update:
                         # Handle read-only
```

Comment:
Should readonly be moved to be a channel-specific setting (essentially the send messages per you've added, for the admin role only)